DEVGUIDE.md

```diff
@@ -0,0 +1,771 @@
+
+```
+    def test_parse_results_takes_failure_percentage_from_parameter_prior_to_config(self):
+        config = {'handler': 'LocustHandler', 'keyword': 'run_locust', 'tags': 'LOCUST', 'failure_percentage': '70'}
+        self.handler = LocustHandler(config)
+        dictionary = self.dictionary_with_result_file()
+        dictionary['failure_percentage'] = 75
+        test_suite = self.handler.parse_results(dictionary)
+        self.assertEqual(test_suite['name'], 'Locust test suite, failure percentage 75')
+
+    def test_parse_results_takes_failure_percentage_correctly_from_config(self):
+        config = {'handler': 'LocustHandler', 'keyword': 'run_locust', 'tags': 'LOCUST', 'failure_percentage': '70'}
+        self.handler = LocustHandler(config)
+        dictionary = self.dictionary_with_result_file()
+        dictionary['failure_percentage'] = None
+        test_suite = self.handler.parse_results(dictionary)
+        self.assertEqual(test_suite['name'], 'Locust test suite, failure percentage 70')
+```
+
+All 8 tests should pass. Now we have completed an LocustHandler with unit tests which test the most important functionalities.
+
+
+
+## How to package your project
+
+Let's package our project in the same virtual environment . Add necessary files defined in this [tutorial](https://packaging.python.org/) to `locustenv` folder and install `setuptools` and `wheel` to your virtualenv with `pip install`. Add following line yo your `setup.py` file in the `setuptools.setup() object`:
+
+```
+    install_requires=[
+           'robotframework-oxygen>=0.1',
+           'locust>=1.1',
+      ],
+```
+
+so that oxygen including it's dependencies and locust will be installed when your handler is installed. Next you can run following command from `locustenv` folder:
+
+```
+python setup.py sdist bdist_wheel
+```
+
+which will create you a `locustenv/dist` folder. Next we will ensure that the installation works by creating another virtualenv. Open up another terminal, go backwards with `cd ..` same path where `locustenv` is and run following commands:
+
+```
+python3 -m venv packagenv
+source packagenv/bin/activate
+pip install locustenv/dist/NAME-OF-YOUR-PACKAGE.whl
+```
+
+you should now have a version of locusthandler in your `packagenv` environment. Let's verify this by opening python intepreter:
+
+```
+python
+import locusthandler.locusthandler
+```
+
+which should succeed. Exit intepreter with CTRL+ D. Next we can add following to the `packagenv/lib/python3.7/site-packages/oxygen/config.yml` file:
+
+```
+locusthandler.locusthandler:
+  handler: LocustHandler
+  keyword: run_locust
+  tags: oxygen-locusthandler
+  failure_percentage: 20
+```
+
+
+ Next let's run the robot test case to make sure that it works.  Copy `test.robot` and `locustfile.py` files to `packagenv/` folder, make the following changes to the variables in `test.robot`:
```

Comment:
the copying of files need a bit clarification because I was confused about why are we copying locusthandler's code since we have already installed it in this place. Maybe something like: "let's make our Locust tests and Robot Framework tests that run them available in our environment." Or something

Reply:
Committed a change for this